tests/rules/test_quoted_strings.py

```diff
@@ -475,6 +476,28 @@ def test_only_when_needed_extras(self):
                    '- "foo bar"\n',
                    conf, problem1=(3, 3), problem2=(7, 3), problem3=(11, 3))
 
+    def test_only_when_needed_special_characters(self):
+        conf = 'quoted-strings: {required: only-when-needed}\n'
+        self.check('---\n'
+                   # double-quoted escaped special chars: ok
+                   'k1: "\\u001b"\n',
+                   conf)
+
+    def test_only_when_needed_special_characters_exceptions(self):
+        conf = 'quoted-strings: {required: only-when-needed}\n'
+        yamltext1 = ('---\n'
+                     # double-quoted unescaped special chars: yuck"
+                     'k1: "\u001b"\n')
+        yamltext2 = ('---\n'
+                     # single-quoted unescaped special chars: yuck"
+                     "k1: '\u001b'\n")
+        yamltext3 = ('---\n'
+                     # unquoted unescaped special chars: yuck"
+                     'k1: \u001b\n')
+        self.assertRaises(yaml_reader_ReaderError, self.check, yamltext1, conf)
+        self.assertRaises(yaml_reader_ReaderError, self.check, yamltext2, conf)
+        self.assertRaises(yaml_reader_ReaderError, self.check, yamltext3, conf)
```

Comment:
Can you please merge the two functions into one? I would find it be more easily understandable that they're about the same issue (like other tests).

Also you can remove all comments `ok / yuck`, we try to avoid those in all tests to be brief.

Reply:
Thanks for the change. Indeed I prefer the new version. I understand the readability argument, but it also applies when reading the whole set of tests: having one single `test_…()` is consistent with other tests for bugfixes, and eases reading yamllint tests globally.